src/annotator/plugin/pdf-metadata.js

```diff
@@ -38,37 +68,40 @@ export default class PDFMetadata {
    */
   constructor(app) {
     /** @type {Promise<PDFViewerApplication>} */
-    this._loaded = new Promise(resolve => {
-      const finish = () => {
-        window.removeEventListener('documentload', finish);
-        window.removeEventListener('documentloaded', finish);
-        app.eventBus?.off('documentloaded', finish);
+    this._loaded = pdfViewerInitialized(app).then(() => {
+      // Check if document has already loaded.
+      if (app.downloadComplete) {
+        return app;
+      }
 
-        resolve(app);
-      };
+      return new Promise(resolve => {
+        const finish = () => {
+          if (app.eventBus) {
+            app.eventBus.off('documentload', finish);
+            app.eventBus.off('documentloaded', finish);
+          } else {
+            window.removeEventListener('documentload', finish);
+          }
+          resolve(app);
+        };
 
-      if (app.downloadComplete) {
-        resolve(app);
-      } else {
         // Listen for "documentloaded" event which signals that the document
         // has been downloaded and the first page has been rendered.
-
-        // Newer versions of PDF.js (>= v2.5.207) report events only via
-        // the PDFViewerApplication's own event bus.
-        app.initializedPromise?.then(() => {
-          app.eventBus?.on('documentloaded', finish);
-        });
-
-        // Older versions of PDF.js (< v2.5.207) dispatch events to the DOM
-        // instead or as well.
-
-        // PDF.js >= v2.0.943.
-        // See https://github.com/mozilla/pdf.js/commit/7bc4bfcc8b7f52b14107f0a551becdf01643c5c2
-        window.addEventListener('documentloaded', finish);
-
-        // PDF.js < v2.0.943.
-        window.addEventListener('documentload', finish);
-      }
+        if (app.eventBus) {
+          // PDF.js >= v1.6.210 dispatch events via an internal event bus.
+          // PDF.js < v2.5.207 also dispatches events to the DOM.
+
+          // `documentloaded` is the preferred event in PDF.js >= v2.0.943.
+          // See https://github.com/mozilla/pdf.js/commit/7bc4bfcc8b7f52b14107f0a551becdf01643c5c2
+          app.eventBus.on('documentloaded', finish);
+
+          // `documentload` is dispatched by PDF.js < v2.1.266.
```

Comment:
Is its meaning different from `documentloaded`? Or was it literally just Very Slightly Renamed (arghghgh)?

Reply:
It is just renamed AFAIK. I haven't checked what the payloads of the events are, but we don't use them anyway.